lib/dataset/bins.cpp

```diff
@@ -380,16 +380,19 @@ Variable bins_sum(const Variable &data) {
 Variable bins_mean(const Variable &data) {
   if (data.dtype() == dtype<bucket<DataArray>>) {
     const auto &&[indices, dim, buffer] = data.constituents<DataArray>();
-    if (const auto mask_union = irreducible_mask(buffer.masks(), dim);
+    if (auto mask_union = irreducible_mask(buffer.masks(), dim);
         mask_union.is_valid()) {
       // Trick to get the sizes of bins if masks are present - bin the masks
       // using the same dimension & indices as the data, and then sum the
       // inverse of the mask to get the number of unmasked entries.
+      mask_union.setUnit(units::one);
```

Comment:
Why is this needed?

Reply:
The unit if the masks is `none`, but we need to normalize to it.

Reply:
Is the reason that `normalize_impl` wants to divide by the sum of masks?

Reply:
Yes. I am just looking into #2224 to see if there is a better solution.